docs/how-to/craft/add-a-snap-configuration.rst

```diff
@@ -0,0 +1,222 @@
+.. _how-to-add-a-snap-configuration:
+
+Add a snap configuration
+========================
+
+This guide describes how to set up :ref:`configuration options
+<explanation-snap-configurations>` in your snap.
+
+
+Review the requirements
+-----------------------
+
+In a standard setup for configuration options, you create four scripts:
+
+- A **wrapper script** calls the app with the necessary environment variables
+- A **management script** sets values, checks their validity, and restarts the app
+- An optional **default-configure hook script** calls the management functions on first
+  install before services are started
+- A **configure hook script** calls the management functions on installation and changes
+
+You must define a configure hook. Otherwise, users can't set any options.
+
+Once the scripts are ready, you link them in the project file.
+
+
+Copy the standard setup
+-----------------------
+
+The following example scripts are for a snap containing an app named ``example-server``.
+They add a configuration option for specifying the server's port.
+
+For a standard setup, copy their implementation into your snap.
+
+
+Add the management script
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A separate script for management functions makes the functions accessible from both the
+wrapper and the configure hook scripts.
+
+The following example defines a default HTTP port and two functions:
+
+- ``http_port`` requests the default port if nothing is yet set and returns the port
+  value
+- ``set_http_port`` sets the port value
+
+Observe how the port itself is obtained from a call to snapctl. It acts as the
+intermediary for all option values.
+
+.. code-block:: bash
+    :caption: src/utilities/bin/management
+
+    #!/bin/sh
+
+    DEFAULT_HTTP_PORT="80"
+
+    http_port()
+    {
+        port="$(snapctl get ports.http)"
+        if [ -z "$port" ]; then
+            port="$DEFAULT_HTTP_PORT"
+            set_http_port $port
+        fi
+        echo "$port"
+    }
+
+    set_http_port()
+    {
+        snapctl set ports.http="$1"
+    }
+
+.. admonition:: Further development
+
+    This script could be expanded to manage the running process, and check whether the
+    new port value is different from the old, saving the service from potentially
+    restarting.
+
+
+Add the default-configure hook script
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The default-configure hook optionally extends the configure hook and executes only on
+snap installation and before services are started.
+
+The following example retrieves a default configuration option from a gadget and either
+writes it to a file, or writes a fallback value if the gadget option doesn't exist:
+
+.. code-block:: bash
+    :caption: src/hooks/bin/default-configure
```

Comment:
@mr-cal I tweaked all the file paths in the scripts and project file to more closely align with the Nextcloud example. Since this isn't a working snap, it hasn't been tested. Please give them a spot-check.